src/System.Private.CoreLib/src/System/Threading/ClrThreadPool.HillClimbing.cs

```diff
@@ -0,0 +1,385 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace System.Threading
+{
+    internal static partial class ClrThreadPool
+    {
+        // Config values pulled from CoreCLR
+        private static readonly HillClimbing s_threadPoolHillClimber = new HillClimbing(4, 20, 100 / 100.0, 8, 15 / 100.0, 300 / 100.0, 4, 20, 10, 200, 1 / 100.0, 200 / 100.0, 15 / 100.0);
+
+        private class HillClimbing
+        {
+            public enum State
```

Comment:
It needs to be public because it is a parameter of `ForceChange`, which is how the thread pool tells hill climbing that it made a mistake and fixes it.